mlir/include/Quantum/IR/QuantumOps.td

```diff
@@ -376,6 +376,74 @@ def CustomOp : UnitaryGate_Op<"custom", [DifferentiableGate, NoMemoryEffect,
         Variadic<QubitType>:$out_ctrl_qubits
     );
 
+    let builders = [
+        OpBuilder<
+        // Convenience builder for a gate with parameters and controls
+        // Note that number of out_qubits = number of in_qubits,
+        // and number of out_ctrl_qubits = number of in_ctrl_qubits
+        (ins
+             "std::string":$gate,
```

Comment:
```suggestion
             "llvm::StringRef":$gate,
```

I found that a lot of times the name of the gate is extracted via `op.getGateName()`

Reply:
done https://github.com/PennyLaneAI/catalyst/pull/1180/commits/199d97d781f0372631b77d72db36a1210578dda0